File: 2000 98 & ME/QuickStart/howto/samples/Remoting/byvalue/VB/object.vb
imports System
imports System.Runtime.Remoting
imports System.Runtime.Remoting.Channels.TCP

namespace RemotingSamples

  public class HelloServer 
    inherits MarshalByRefObject

    public sub HelloServer()
      Console.WriteLine("HelloServer activated")
    end sub


    public function HelloMethod(obj as ForwardMe) as ForwardMe 
      
      dim i as integer
      
      for i = 0 to 5
        obj.CallMe()
      next i
        
      HelloMethod = obj
    end function
    
  end class
  
end namespace

File: 2000 98 & ME/QuickStart/howto/samples/Remoting/byvalue/object.cs
using System;
using System.Runtime.Remoting;
using System.Runtime.Remoting.Channels.TCP;

namespace RemotingSamples {
  public class HelloServer : MarshalByRefObject {

    public HelloServer()
    {
      Console.WriteLine("HelloServer activated");
    }


    ~HelloServer()
    {
      Console.WriteLine("Object Destroyed");
    }


    public ForwardMe HelloMethod(ForwardMe obj)
    {
      for (int i=0; i < 5; i++) obj.CallMe();
      return obj;
    }
  }
}
